### ISSUE TITLE
Performance improvements

### ISSUE THREAD

**Issue by phips28** [timgit/pg-boss#31]
Hi,

we are currently testing pg-boss on a high volume event trigger machine. 1000 jobs added/sec + 10*80 worker.
And we see the SELECT to get the job is very unperformant without an index.
Adding following indexes improved it: pg cost 50000 => 14

```
CREATE INDEX job_createdon_id ON shard_01_pgboss.job (createdon, id); -- very important!!!
CREATE INDEX job_state ON shard_01_pgboss.job (state); -- sometimes pg query optimizer takes this index; e.g. if there are close to zero 'created' jobs
```

What was your highest tested jobs/sec?

**Comment by phips28** [comment]
Ah and I see for the new branch:1.2 you would need a different index with `priority`.

**Comment by timgit** [comment]
This is great feedback!  I've been waiting for your report.  👍 😉 You beat me to this level of load testing, as our product is just now beginning to utilize pgboss for a much larger scale than I originally wrote it for.

I'm totally cool with experimenting with different indexes, your examples being an excellent baseline. Indexes are the best for handling a large, unexpected load.  If the load is known ahead of time, another strategy at our disposal is setting an archive threshold and interval to keep record counts lower.

**Comment by phips28** [comment]
We had a little issue with our setup. Overnight we created more jobs as the worker could handle.
In this case it grew to 200k jobs 'created', and to get one newJob it too ~1-2sec (because of the missing index) 
But now we figured it out how to do it right ;)

**Comment by timgit** [comment]
In regards to 1.2, it probably won't be released as I've rolled all those changes into a 2.0 release.

**Comment by phips28** [comment]
ok, then you need to add the index to 2.0 ;)

**Comment by timgit** [comment]
Added to 2.0.0-rc1

**Comment by timgit** [comment]
Released with 2.0.0
